src/hooks/useIsVisible.jsx

```diff
@@ -5,16 +5,24 @@ const useIsVisible = (defaultIsVisible = true) => {
   const [isVisible, setIsVisible] = useState(defaultIsVisible);
 
   useEffect(() => {
-    if (sentinelRef.current) {
-      const observer = new IntersectionObserver((entries) => {
-        entries.forEach(({ isIntersecting }) => {
-          setIsVisible(isIntersecting);
-        });
-      }, {});
-      observer.observe(sentinelRef.current);
-      return () => {
-        observer.disconnect();
-      };
+    try {
+      if (sentinelRef.current) {
+        const observer = new IntersectionObserver((entries) => {
+          entries.forEach(({ isIntersecting }) => {
+            setIsVisible(isIntersecting);
+          });
+        }, {});
+        observer.observe(sentinelRef.current);
+        return () => {
+          observer.disconnect();
+        };
+      }
+    } catch (e) {
```

Comment:
Can it not just be `catch(isReferenceError) `? Is that too sensible for Javascript?

Reply:
No, the `(e)` is the error itself, more like a function definition than an if statement https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch